cpp/src/neighbors/batch_ann/batch_ann.cuh

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include "../detail/ann_utils.cuh"
```

Comment:
Everytime we include (and use) detail code, it means any kernels or device code has to be recompiled, which increases compile time and binary size. We should be very careful to make sure we're only using public APIs as much as possible, especially when we're building things on top like this API is doing. There's no actual public API for ann_utils, so my hope is that you aren't actually using it and we can just remove this include.

Reply:
Yes, I'm not sure why it slipped in...